packit/api.py

```diff
@@ -362,6 +390,161 @@ def update_source_git(self, revision_range: str):
                     f"Commit {commit} had no changes to be applied, skipping it."
                 )
 
+    def show_sync_status(
+        self, status: SynchronizationStatus, source_git: str, dist_git: str
+    ) -> None:
+        """Outputs the synchronization status of source-git and dist-git.
+
+        Args:
+            status: Synchronization status of source-git and dist-git.
+            source_git: Path to source-git as specified by the user.
+            dist_git: Path to dist-git as specified by the user.
+        """
+        if status.source_git_range_start and status.dist_git_range_start:
+            click.echo(f"'{source_git}' and '{dist_git}' have diverged.")
+            click.echo("Sync status needs to be reestablished manually.")
+            click.echo(
+                f"First unsynced commit in source-git: {status.source_git_range_start}."
+            )
+            click.echo(
+                f"First unsynced commit in dist-git: {status.dist_git_range_start}."
+            )
+        elif status.source_git_range_start:
+            number_of_commits = len(
+                list(
+                    self.up.local_project.git_repo.iter_commits(
+                        f"{status.source_git_range_start}~..", ancestry_path=True
+                    )
+                )
+            )
+            click.echo(
+                f"'{source_git}' is ahead of '{dist_git}' by {number_of_commits} commits."
+            )
+            click.echo(
+                f'Use "packit-dev source-git update-dist-git {source_git} {dist_git}" '
+                f"to transform changes from '{source_git}' to '{dist_git}'."
+            )
+            click.echo(
+                f"First unsynced commit in source-git: {status.source_git_range_start}."
+            )
+        elif status.dist_git_range_start:
+            number_of_commits = len(
+                list(
+                    self.dg.local_project.git_repo.iter_commits(
+                        f"{status.dist_git_range_start}~..", ancestry_path=True
+                    )
+                )
+            )
+            short_hash = self.dg.local_project.git_repo.git.rev_parse(
+                status.dist_git_range_start, short=True
+            )
+            click.echo(
+                f"'{source_git}' is behind of '{dist_git}' by {number_of_commits} commits."
+            )
+            click.echo(
+                f'Use "packit-dev source-git update-source-git {dist_git} {source_git} '
+                f"{short_hash}~..\" to transform changes from '{dist_git}' to '{source_git}'."
```

Comment:
And then you can get rid of the `rev-parse` call above.

```suggestion
                f"{status.dist_git_range_start[:7]}~..\" to transform changes from '{dist_git}' to '{source_git}'."
```